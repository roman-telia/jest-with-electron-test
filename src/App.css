@import url('https://teliads-assets.netlify.app/fonts/index.css');
.App {
  text-align: center;
  font-family: 'TeliaSans';
}
.pebble {
  font-family: 'Pebble';
} 
