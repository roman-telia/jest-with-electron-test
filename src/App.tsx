import React from 'react';
import logo from './logo.svg';
import {TeliaButton, TeliaCol, TeliaGrid, TeliaRow, TeliaHeading, TeliaP, TeliaLink} from '@teliads/components/react';
import './App.css';

function App() {
  return (
    <div className="App">
      <TeliaGrid>
        <TeliaRow>
        <TeliaCol width={12}><TeliaHeading variant="display-200" tag="h1">Display 200 H1</TeliaHeading></TeliaCol>
        <TeliaCol width={12}><TeliaHeading variant="display-100" tag="h2">Display 100 H2</TeliaHeading></TeliaCol>
        <TeliaCol width={6}><TeliaHeading variant="title-100" tag="h3">Hi hello Hola</TeliaHeading></TeliaCol>
        <TeliaCol width={6}><TeliaHeading variant="title-200" tag="h3">Hi hello Hola</TeliaHeading></TeliaCol>
        <TeliaCol width={6}><TeliaHeading variant="title-300" tag="h3">Hi hello Hola</TeliaHeading></TeliaCol>
        <TeliaCol width={6}><TeliaHeading variant="title-400" tag="h3">Hi hello Hola</TeliaHeading></TeliaCol>
        <TeliaCol width={6}><TeliaHeading variant="title-500" tag="h3">Hi hello Hola</TeliaHeading></TeliaCol>
        <TeliaCol width={6}><TeliaHeading variant="title-600" tag="h3">Hi hello Hola</TeliaHeading></TeliaCol>
        <TeliaCol width={6}><TeliaHeading variant="title-700" tag="h3">Hi hello Hola</TeliaHeading></TeliaCol>
        </TeliaRow>
      </TeliaGrid>
      <TeliaButton variant="primary" id="telia-button" dataTestid="telia-button" children="Hi there"></TeliaButton>
      <TeliaP>loerm sddds dss sswew <TeliaLink href="https://www.telia.fi" variant="silent">Here</TeliaLink></TeliaP>
      
    </div>
  );
}

export default App;
