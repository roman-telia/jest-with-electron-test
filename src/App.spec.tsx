
// import {TeliaButton as TeliaButtonElement} from '@teliads/components/react';
// import {TeliaButton} from '@teliads/components/dist/types/components/telia-button/telia-button';
// describe('telia-button', () => {

//     const AWESOME_BUTTON_TAG = 'telia-button';
//     const ELEMENT_ID = 'telia-button';
//     let buttonElement: HTMLTeliaButtonElement ;

//     const getShadowRoot = (tagName: string): ShadowRoot => {
//         return document.body.getElementsByTagName(tagName)[0].shadowRoot as ShadowRoot;
//     }

//     beforeEach(() => {
//         buttonElement = window.document.createElement(AWESOME_BUTTON_TAG) as HTMLTeliaButtonElement;
//         document.body.appendChild(buttonElement);
//         console.log(document.body.)
//     });

//     afterEach(() => {
//        document.body.getElementsByTagName(AWESOME_BUTTON_TAG)[0].remove();
//     });

//     it('displays button text', async () => {
//         const dummyText = 'Web components & Jest with Electron';
//         buttonElement.setAttribute('children', dummyText);
//         await buttonElement.componentOnReady;

//         const renderedText = getShadowRoot(AWESOME_BUTTON_TAG).getElementById(ELEMENT_ID)?.innerText;

//         expect(renderedText).toEqual(dummyText);
//     });
//     it('handles clicks', async () => {
//         const mockClickFunction = jest.fn();
//         buttonElement.addEventListener('click', () => {mockClickFunction()});

//         getShadowRoot(AWESOME_BUTTON_TAG).getElementById(ELEMENT_ID)?.click();
//         getShadowRoot(AWESOME_BUTTON_TAG).getElementById(ELEMENT_ID)?.click();

//         expect(mockClickFunction).toHaveBeenCalledTimes(2);
//     });
// });

import { spy } from 'sinon';
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import App from './App';
import {TeliaColorDot} from '@teliads/components/react/commonjs';

Enzyme.configure({ adapter: new Adapter() });
function setup() {
  const actions = {
    increment: spy(),
    incrementIfOdd: spy(),
    incrementAsync: spy(),
    decrement: spy()
  };
  const component = shallow(<TeliaColorDot color="pink" withborder={true} />);
  return {
    component,
    actions,
    buttons: component.find('div'),
  };
}

// --snip--

describe('App Component', () => {
  it('should match exact snapshot', () => {
    const { actions } = setup();
    const counter = (
      <TeliaColorDot color="pink" withborder={true} />
    );
    const tree = renderer.create(counter).toJSON();

    expect(tree).toMatchSnapshot();
  });
});