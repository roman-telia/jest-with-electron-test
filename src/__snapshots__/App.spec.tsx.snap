// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should match exact snapshot 1`] = `
<telia-color-dot
  color="pink"
  withborder={true}
/>
`;
